### PR TITLE
Upgrade react-syntax-highlighter in presentational-components (again, with less problems, hopefully)

### PR DESCRIPTION
**Pull request by akx**
Ugh, sorry about the mess I brought on in #4636 (refs #4663, #4664).
I was actually bit by it personally too, now that the upgraded module was published. 😅 

This is essentially the same PR as #4636, but without ESM imports; I just vendored in those troublesome theme modules instead.

### CODE REVIEW

**Comment by wadethestealth** [comment]
Before merging, I need to test that this would work in Hydrogen. ~I will update this comment with the results.~

~Results: https://github.com/nteract/nteract/pull/4669#discussion_r341831963~ My mistake.
Results: works and specs pass

File: packages/presentational-components/src/syntax-highlighter/index.tsx [hunk 1]
```diff
@@ -1,11 +1,7 @@
 import * as React from "react";
-import SyntaxHighlighter from "react-syntax-highlighter/prism";
-import {
-  vs as lightTheme,
-  xonokai as darkTheme
-} from "react-syntax-highlighter/styles/prism";
-
-// import syntax from "./theme";
+import {Prism as SyntaxHighlighter} from "react-syntax-highlighter";
```
Comment (rgbkrk) [inline packages/presentational-components/src/syntax-highlighter/index.tsx:2]: One way of guaranteeing this working is by using the [direct cjs import the library authors recommend (though stated as non-ideal)](https://github.com/conorhastings/react-syntax-highlighter/blob/d58813a0f2e2fd4ba94e5c1122e1ddb9ca01f67f/CHANGELOG.MD#changed-1):

```
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter/dist/cjs/prism";
```

Comment (wadethestealth) [inline packages/presentational-components/src/syntax-highlighter/index.tsx:2]: Currently, this dosn't even build. To test builds please run `tsc -b packages/presentational-components` to test bulding
```
packages/presentational-components/src/syntax-highlighter/index.tsx:2:9 - error TS2305: Module '"react-syntax-highlighter"' has no exported member 'Prism'.

2 import {Prism as SyntaxHighlighter} from "react-syntax-highlighter";
```

**Review by wadethestealth — CHANGES_REQUESTED** [review]

File: packages/presentational-components/src/syntax-highlighter/index.tsx (hunk 1 above)
Comment (akx) [reply packages/presentational-components/src/syntax-highlighter/index.tsx:2]: Wait, um, I'm not seeing that (and CI passes too)...

```shell
$ git rev-parse HEAD
b5ed3382ab2ea9085d203a1e6c85d2e82ba67150
$ ./node_modules/.bin/tsc -b packages/presentational-components
$
```

Comment (wadethestealth) [reply packages/presentational-components/src/syntax-highlighter/index.tsx:2]: The ci passes because this module has no tests

Comment (wadethestealth) [reply packages/presentational-components/src/syntax-highlighter/index.tsx:2]: I think that was my mistake though let me try again.

Comment (akx) [reply packages/presentational-components/src/syntax-highlighter/index.tsx:2]: `react-syntax-highlighter`'s `package.json` has 
```json
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
```
so bundlers/runtimes will know to use the ESM build if they're ESM aware, and use the CJS build otherwise.

The problem was with the direct imports of `esm` style theme modules which didn't get transpiled and thus were not importable.

As for the theme modules, sure, they could be imported `cjs` style, so they wouldn't need to get bundled, at the likely expense of a little more boilerplate in the code here. I can do that, too, if that's preferred over vendoring the themes.

Comment (wadethestealth) [reply packages/presentational-components/src/syntax-highlighter/index.tsx:2]: Ok that was totally my fault. It does build, but we do need to be careful because the ci does not test this module

Comment (wadethestealth) [reply packages/presentational-components/src/syntax-highlighter/index.tsx:2]: I think importing a cjs rather then updating a vender is better.

Comment (akx) [reply packages/presentational-components/src/syntax-highlighter/index.tsx:2]: Okiedoke, changed that. Did mean I had to add an "ad-hoc" `.d.ts` file for the two modules, but it's explained in the commit message.

**Review by wadethestealth — APPROVED** [review]

**Comment by akx** [comment]
Thanks! Here's hoping I don't break anything this time...
